server/routes/bookings/bookingDetailsController.ts

```diff
@@ -30,12 +30,17 @@ export default class BookingDetailsController {
 
       const prison = await this.prisonService.getPrison(visit.prisonId)
 
+      const nowTimestamp = new Date()
+      const visitStartTimestamp = new Date(visit.startTimestamp)
+      const showCancel = nowTimestamp < visitStartTimestamp
+
       return res.render('pages/bookings/visit', {
         booker,
         prison,
         prisoner,
         type,
         visit,
+        showCancel,
```

Comment:
Could add to `bookingDetailsController.test.ts` to assert that cancel button **not** present on past/cancelled details view and **is** present on future details view.